src/main/java/com/faforever/api/config/GlobalControllerExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.faforever.api.config;
+
+import com.faforever.api.config.error.ErrorResponse;
+import com.faforever.api.config.error.ErrorResult;
+import com.faforever.api.error.ApiException;
+import com.faforever.api.error.ErrorCode;
+import com.google.common.collect.ImmutableMap;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import javax.validation.ValidationException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Map;
+
+@ControllerAdvice
+class GlobalControllerExceptionHandler {
+  private Map<String, Serializable> errorResponse(String title, String message) {
+    ImmutableMap<String, Serializable> error = ImmutableMap.of(
+        "title", title,
+        "detail", message);
+    return ImmutableMap.of("errors", new ImmutableMap[]{error});
+  }
+
+  @ExceptionHandler(ValidationException.class)
+  @ResponseStatus(HttpStatus.UNPROCESSABLE_ENTITY)
+  @ResponseBody
+  public Map<String, Serializable> processValidationException(ValidationException ex) {
+    return errorResponse(ErrorCode.VALIDATION_FAILED.getTitle(), ex.getMessage());
+  }
+
+  private String replaceArgs(String message, Object[] args) {
+    for (int i = 0; i < args.length; i++) {
+      message = message.replace("{" + i + "}", args[i].toString());
```

Comment:
* If `args[i]` is `null` (for whatever reason) this throws an NPE, which is why the null-safe `String.valueOf(Object)` should be used instead.
* However, `{0}` and such is part of the Java MessageFormat, which is why `MessageFormat.format(String, Object...)` should be used. So this method should be removed (MessageFormat does much more than just replace {n}, too)